app/assets/javascripts/discourse/tests/integration/components/user-menu/menu-test.js

```diff
@@ -49,7 +70,162 @@ module("Integration | Component | user-menu", function (hooks) {
     assert.strictEqual(tabs.length, 1);
     const preferencesTab = tabs[0];
     assert.ok(preferencesTab.href.endsWith("/u/eviltrout/preferences"));
-    assert.strictEqual(preferencesTab.getAttribute("data-tab-number"), "1");
+    assert.strictEqual(preferencesTab.getAttribute("data-tab-number"), "4");
     assert.strictEqual(preferencesTab.getAttribute("tabindex"), "-1");
   });
+
+  test("likes tab is hidden if current user's like notifications frequency is 'never'", async function (assert) {
+    this.currentUser.set("likes_notifications_disabled", true);
+    await render(template);
+    assert.ok(!exists("#user-menu-button-likes"));
+
+    const tabs = Array.from(queryAll(".tabs-list .btn")); // top and bottom tabs
+    assert.strictEqual(tabs.length, 4);
+
+    assert.deepEqual(
+      tabs.map((t) => t.getAttribute("data-tab-number")),
```

Comment:
That's a nitpick but generally I prefer to use the dataset for this as it's the dedicated API: `t.dataset.tabNumber`